src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -308,12 +431,21 @@ func (sm *StorageMinerActor_I) checkIfSectorExists(sectorNo sector.SectorNumber)
 // Currently deals must be posted on chain via sma.PublishStorageDeals before CommitSector
 // TODO(optimization): CommitSector could contain a list of deals that are not published yet.
 func (sm *StorageMinerActor_I) CommitSector(onChainInfo sector.OnChainSealVerifyInfo) {
+
+	if sm.isChallenged() {
+		// TODO: proper throw
+		panic("cannot CommitSector when sm isChallenged")
+	}
```

Comment:
we should be able to commit sectors while in challenge period -- this will be very annoying operationally if not. does the frozen proving set work well, or no?

Reply:
Given the current set up, freezing ProvingSet with out freezing SectorTable and Sectors wont work because we need to know the SectorTable breakdown within the ProvingSet to account for power and collateral. We are freezing all three of them with isChallenged(). I can add a separate commit that allows for CommitSector while isChallenged() and keep those sectors as a buffer. That buffer will be added to the ProvingSet and SectorTable upon successful SubmitPoSt or MissedPoSt. We can decide if we want to include this later.